meson.build

```diff
@@ -0,0 +1,171 @@
+endif
+if magick_dep.found()
+    if magick_dep.version().version_compare('>=7.0')
+        cfg_var.set('HAVE_MAGICK7', '1')
+    elif magick_dep.version().version_compare('>=6.0')
+        cfg_var.set('HAVE_MAGICK6', '1')
+        if cc.has_member('struct _ImageInfo', 'number_scenes', prefix : '#include <magick/api.h>', dependencies: magick_dep)
+            cfg_var.set('HAVE_NUMBER_SCENES', '1')
+        endif
+        func_names = [ 'InheritException', 'AcquireExceptionInfo', 'SetImageProperty', 'SetImageExtent', 'AcquireImage', 'GetVirtualPixels', 'ResetImageProfileIterator', 'ResetImageAttributeIterator', 'ResetImagePropertyIterator', 'MagickCoreGenesis', 'SetImageOption', 'BlobToStringInfo', 'OptimizePlusImageLayers', 'OptimizeImageTransparency' ]
+        foreach func_name : func_names
+            if not cc.has_function(func_name, dependencies: magick_dep)
+                error('Unsupported libmagick version: missing ' + func_name)
+            endif
+        endforeach
+        if cc.compiles('#include <magick/api.h>\nColorspaceType colorspace = CMYColorspace;', dependencies: magick_dep)
+            cfg_var.set('HAVE_CMYCOLORSPACE', '1')
+        endif
+        if cc.compiles('#include <magick/api.h>\nColorspaceType colorspace = HCLpColorspace;', dependencies: magick_dep)
+            cfg_var.set('HAVE_HCLPCOLORSPACE', '1')
+        endif
+        # GetImageMagick() takes two args under GM, three under IM
+        if cc.compiles('#include <magick/api.h>\n(void)GetImageMagick(NULL, 0, NULL);', dependencies: magick_dep)
+            cfg_var.set('HAVE_GETIMAGEMAGICK3', '1')
+        endif
+    else
+        error('Unsupported libmagick version')
+    endif
+    #
+endif
+
+if cc.has_header('sys/file.h')
+    cfg_var.set('HAVE_SYS_FILE_H', '1')
+endif
+if cc.has_header('sys/param.h')
+    cfg_var.set('HAVE_SYS_PARAM_H', '1')
+endif
+if cc.has_header('sys/mman.h')
+    cfg_var.set('HAVE_SYS_MMAN_H', '1')
+endif
+if cc.has_header('unistd.h')
+    cfg_var.set('HAVE_UNISTD_H', '1')
+endif
+if cc.has_header('io.h')
+    cfg_var.set('HAVE_IO_H', '1')
+endif
+if cc.has_header('direct.h')
+    cfg_var.set('HAVE_DIRECT_H', '1')
+endif
+if cc.has_header('windows.h')
+    cfg_var.set('HAVE_WINDOWS_H', '1')
+endif
+if get_option('deprecated')
+    cfg_var.set('ENABLE_DEPRECATED', '1')
+endif
+if get_option('nsgif')
+    cfg_var.set('HAVE_NSGIF', '1')
+endif
+if get_option('ppm')
+    cfg_var.set('HAVE_PPM', '1')
+endif
+if get_option('analyze')
+    cfg_var.set('HAVE_ANALYZE', '1')
+endif
+if get_option('deprecated')
+    cfg_var.set('ENABLE_DEPRECATED', '1')
+endif
+
+#AC_CHECK_HEADERS([])
+cfg_var.set_quoted('GETTEXT_PACKAGE', 'vips8.13')
+cfg_var.set_quoted('VIPS_PREFIX', get_option('prefix'))
+cfg_var.set_quoted('VIPS_LIBDIR', get_option('prefix') / get_option('libdir'))
+libintl_dep = cc.find_library('intl', required : false)
+if libintl_dep.found()
+    cfg_var.set('ENABLE_NLS', 1)
```

Comment:
This will not be found on platforms where the gettext family of functions is built into the libc. That means it won't work on Linux with either the GNU libc or musl libc, and won't work on any of the BSDs by default.

It will be found on Windows using MSYS2, if GNU libintl is installed, or on the BSDs in the event that the user manually installs the GNU libintl port.

Instead, do one of two things:
- check if `cc.has_function('ngettext')` is successfully found, and if it is not found, fallback to checking if the `intl` library can be found
- Use https://mesonbuild.com/Dependencies.html#intl, which will just report "not found" on versions of meson before 0.59

Reply:
Thanks for the hint, I've adapted it to the first suggestion, I hope that we could just use `glib/gi18n.h` at some point as we already have GLib available